Jetpack Agency Dashboard: set container element's height to 100%

rcanepa opened the pull request:
#### Proposed Changes

* Currently, a div's height is set to `100vh`, which extends the page's height unnecessarily (we end up with a vertical scroll bar). These changes fix that by setting its parent component's height to `100vh` and replacing the child's height from `100vh` to `100%`. As a result, the page no longer shows the vertical scroll bar.

<table>
<tr>
<th>
Before
</th>
<th>
After
</th>
</tr>
<tr>
<td>
<img width="1462" alt="Screen Shot 2022-09-05 at 11 50 43" src="https://user-images.githubusercontent.com/3418513/188485572-870660b0-5c35-4e61-9053-017d77381347.png">
</td>
<td>
<img width="1462" alt="Screen Shot 2022-09-05 at 11 50 53" src="https://user-images.githubusercontent.com/3418513/188485532-fe91dc44-d026-47bd-8386-2aca5042236c.png">
</td>
</tr>
</table>


#### Testing Instructions

##### Replicate the issue
* Visit https://cloud.jetpack.com/dashboard.
* On the desktop, confirm you see the vertical scroll bar on the right of the page (this happens regardless of the amounts of sites shown in the dashboard).

##### Test the fix
* Download this PR.
* Open http://jetpack.cloud.localhost:3001/, and you'll be redirected to the /dashboard.
* On the desktop, confirm you don't see the vertical scroll bar on the right of the page.
* Verify that everything still looks correct under different viewport sizes.

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to 1202518759611394-as-1202781858630727

rcanepa committed:
Set 100vh to container element

rcanepa committed:
Include paddings in the height computation

rcanepa commented:
FWIW: the failed check is unrelated to these changes.

vitozev reviewed (commented):
I think we need to apply the same styles for the plugin management container?

If you visit http://jetpack.cloud.localhost:3000/plugins/updates you will see the same issue.

yashwin reviewed (approved):
LGTM. I ran the test again, and it passed. 

I agree with Kristian's comment here, we also have a similar issue on the `Plugins` page.

rcanepa committed:
Update height and margin in plugin area

rcanepa committed:
Set the right height with flex

rcanepa commented:
@yashwin, @vitozev, I updated the PR to include a fix to the plugins section. I changed the approach a little bit (now I'm using `display:flex` to fix it instead of `height`).

I would appreciate another round of reviews.

vitozev reviewed (approved):
LGTM!
